rust/kernel/hw_random.rs

```diff
@@ -0,0 +1,200 @@
+// SPDX-License-Identifier: GPL-2.0
+
+//! Hardware Random Number Generator.
```

Comment:
I think the name of this file should be `hwrng` because that's the prefix that the kernel uses for its functions. So one could just import it, and use types like `hwrng::Registration`.

Reply:
Makes sense, thanks.